File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        indexT, found=0, 0
        for x in s:
            while indexT<len(t):
                if t[indexT]==x:
                    found+=1
                    indexT+=1
                    break
                indexT+=1
        if found==len(s):
            return True
        return False

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.length()==0)
            return true;
        int i=0;
        for (auto x: t)
        {
            if (x==s[i])
            {
                i++;
                if (i==s.length())
                {
                    return true;
                }
            }
        }
        return false;
    }
};